lib/utils/isDescriptorDeclaration.mjs

```diff
@@ -0,0 +1,17 @@
+import { isAtRule } from './typeGuards.mjs';
+import { nestingSupportedAtKeywords } from '../reference/atKeywords.mjs';
+
+/**
+ * Check whether a declaration is a descriptor one
+ *
+ * @param {import('postcss').Declaration} decl
+ * @returns {boolean}
+ */
+export default function isDescriptorDeclaration(decl) {
+	const { parent } = decl;
+
+	if (parent && isAtRule(parent) && !nestingSupportedAtKeywords.has(parent.name.toLowerCase()))
```

Comment:
question

if
```js
@position-try --custom-bottom-right {
  @supports (position-area: bottom right) {
    position-area: bottom right;
  }
}
```
is allowed this `if` won't be enough.
i.e. you will probably have to check all its parents which will probably degrade the perf

Reply:
Good catch, thanks!

I've pushed a commit to address. Let's favour accuracy over performance in this instance.